internal/controller/persistentvolumeclaim_controller_test.go

```diff
@@ -555,4 +555,98 @@ var _ = Describe("PersistentVolumeClaimController controller", func() {
 			}), "when delete "+pvcMeta.Namespace+"/"+pvcMeta.Name)
 		}
 	})
+
+	It("should annotate the Pod to notify kubelet to resize the filesystem when the PVC has FileSystemResizePending condition", func() {
+		ctx := context.Background()
+		ns := createNamespace()
+		pod := corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pod0",
+				Namespace: ns,
+			},
+			Spec: corev1.PodSpec{
+				Containers: []corev1.Container{
+					{
+						Name:  "container",
+						Image: "registry.k8s.io/pause",
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      "vol0",
+								MountPath: "/vol0",
+							},
+						},
+					},
+				},
+				Volumes: []corev1.Volume{
+					{
+						Name: "vol0",
+						VolumeSource: corev1.VolumeSource{
+							PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+								ClaimName: "pvc0",
+							},
+						},
+					},
+				},
+			},
+		}
+		pvc := corev1.PersistentVolumeClaim{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "pvc0",
+				Namespace: ns,
+				Finalizers: []string{
+					"topolvm.io/pvc",
+				},
+			},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					corev1.ReadWriteOnce,
+				},
+				Resources: corev1.VolumeResourceRequirements{
+					Requests: corev1.ResourceList{
+						"storage": resource.MustParse("1Gi"),
+					},
+				},
+			},
+		}
+
+		err := k8sClient.Create(ctx, &pod)
+		Expect(err).NotTo(HaveOccurred())
+		err = k8sClient.Create(ctx, &pvc)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("Checking a precondition that the Pod doesn't have annotation")
+		Eventually(func(g Gomega) {
+			var got corev1.Pod
+			err := k8sClient.Get(ctx, client.ObjectKey{Namespace: pod.Namespace, Name: pod.Name}, &got)
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(got.Annotations).ShouldNot(HaveKey("topolvm.io/resizefs-requested-at"))
+		}).Should(Succeed())
```

Comment:
```suggestion
		}).WithContext(ctx).Should(Succeed())
```
I think its best practice to pass the context to all eventuallies: https://onsi.github.io/ginkgo/#using-speccontext-with-gomegas-eventually